src/ru/nsu/ccfit/zuev/skins/OsuSkin.java

```diff
@@ -0,0 +1,155 @@
+    protected final BooleanSkinData rotateCursor = new BooleanSkinData("rotateCursor", true);
+
+    protected final String DEFAULT_COLOR_HEX = "#FFFFFF";
+    protected final RGBColor DEFAULT_COLOR = RGBColor.hex2Rgb(DEFAULT_COLOR_HEX);
+    protected final ArrayList<RGBColor> comboColor = new ArrayList<>();
+
+    protected final ColorSkinData sliderBorderColor = new ColorSkinData("sliderBorderColor", DEFAULT_COLOR);
+    protected final ColorSkinData sliderBodyColor = new ColorSkinData("sliderBodyColor", DEFAULT_COLOR);
+    protected final ColorSkinData sliderHintColor = new ColorSkinData("sliderHintColor", DEFAULT_COLOR);
+
+    protected final SkinSliderType skinSliderType = SkinSliderType.FLAT;
+
+    protected final HashMap<String, SkinLayout> layoutData = new HashMap<>();
+    protected final HashMap<String, RGBColor> colorData = new HashMap<>();
+
+    public static OsuSkin get() {
+        return skinJson;
+    }
+
+    @NonNull
+    public static String readFull(File file) throws IOException {
+        BufferedSource source = Okio.buffer(Okio.source(file));
+        String result = source.readUtf8();
+        source.close();
+        return result;
+    }
+
+    public boolean isRotateCursor() {
+        return rotateCursor.getCurrentValue();
+    }
+
+    public float getComboTextScale() {
+        return comboTextScale.getCurrentValue();
+    }
+
+    public boolean isUseNewLayout() {
+        return useNewLayout.getCurrentValue();
+    }
+
+    public boolean isSliderHintEnable() {
+        return sliderHintEnable.getCurrentValue();
+    }
+
+    public float getSliderHintAlpha() {
+        return sliderHintAlpha.getCurrentValue();
+    }
+
+    public float getSliderHintWidth() {
+        return sliderHintWidth.getCurrentValue();
+    }
+
+    public RGBColor getSliderHintColor() {
+        return sliderHintColor.getCurrentValue();
+    }
+
+    public float getSliderHintShowMinLength() {
+        return sliderHintShowMinLength.getCurrentValue();
+    }
+
+    public float getSliderBodyWidth() {
+        return sliderBodyWidth.getCurrentValue();
+    }
+
+    public float getSliderBorderWidth() {
+        return sliderBorderWidth.getCurrentValue();
+    }
+
+    public boolean isDisableKiai() {
+        return disableKiai.getCurrentValue();
+    }
+
+    public boolean isLimitComboTextLength() {
+        return limitComboTextLength.getCurrentValue();
+    }
+
+    public float getSliderBodyBaseAlpha() {
+        return sliderBodyBaseAlpha.getCurrentValue();
+    }
+
+    public boolean isForceOverrideComboColor() {
+        return forceOverrideComboColor.getCurrentValue();
+    }
+
+    public ArrayList<RGBColor> getComboColor() {
+        if (comboColor.isEmpty()) {
+            comboColor.add(DEFAULT_COLOR);
+        }
+        return comboColor;
+    }
+
+    public boolean isForceOverrideSliderBorderColor() {
+        return sliderBorderColor.getCurrentValue() != sliderBorderColor.getDefaultValue();
```

Comment:
This comparison will not work and always return `false`. You cannot directly compare classes with logical operators.

Reply:
Your most recent commit will make beatmap-specific slider border colors not work since you're overriding it with skin-specific/default slider border color. It's not solving the problem pointed in here.